packages/core/src/modules/oob/OutOfBandModule.ts

```diff
@@ -0,0 +1,297 @@
+          }
+        } else {
+          this.logger.debug('Reuse is disabled. Creating a new connection.')
+          connectionRecord = await this.createConnection(outOfBandMessage, autoAccept)
+        }
+      } else {
+        this.logger.debug('Connection does not exists. Creating a new connection.')
+        connectionRecord = await this.createConnection(outOfBandMessage, autoAccept)
+      }
+
+      if (messages) {
+        this.logger.debug('Out of band message contains request messages.')
+        if (!connectionRecord.isReady) {
+          // Wait until the connecion is ready and then pass the messages to the agent for further processing
+          this.connectionsModule
+            .returnWhenIsConnected(connectionRecord.id)
+            .then((c) => {
+              const connectionServices = c.theirDidDoc?.didCommServices
+              this.emitMessages(connectionServices, messages)
+            })
+            .catch((error) => {
+              if (error instanceof EmptyError) {
+                this.logger.warn(
+                  `Agent unsubscribed before connection got into ${ConnectionState.Complete} state`,
+                  error
+                )
+              } else {
+                this.logger.error('Promise waiting for the connection to be complete failed.', error)
+              }
+            })
+        } else {
+          const connectionServices = connectionRecord.theirDidDoc?.didCommServices
+          this.emitMessages(connectionServices, messages)
+        }
+      }
+
+      return connectionRecord
+    } else if (messages) {
+      this.emitMessages(services, messages)
+    }
+  }
+
+  private areHandshakeProtocolsSupported(handshakeProtocols: string[]) {
+    return this.getSupportedHandshakeProtocols().some((p) => handshakeProtocols.includes(p))
+  }
+
+  private getSupportedHandshakeProtocols() {
+    const handshakeMessageFamilies = ['https://didcomm.org/didexchange', 'https://didcomm.org/connections']
+    const handshakeProtocols = this.dispatcher.filterSupportedProtocolsByMessageFamilies(handshakeMessageFamilies)
+
+    if (handshakeProtocols.length === 0) {
+      throw new AriesFrameworkError('There is no handshake protocol supported. Agent can not create a connection.')
+    }
+
+    return handshakeProtocols
+  }
+
+  private async findExistingConnection(services: DidCommService[]) {
+    for (const service of services) {
+      for (const recipientKey of service.recipientKeys) {
+        // TODO Encode the key and endpoint of the service block in a Peer DID numalgo 2 and using that DID instead of a service block
+        const existingConnection = await this.connectionsModule.findByTheirKey(recipientKey)
+        return existingConnection
+      }
+    }
+  }
+
+  private async createConnection(outOfBandMessage: OutOfBandMessage, autoAccept: boolean) {
+    const { services, label } = outOfBandMessage
+    const invitation = new ConnectionInvitationMessage({
+      label: label || '',
```

Comment:
To my understanding `label` is not optional, either way we shouldn't use `''`

Reply:
[OOB spec](https://github.com/hyperledger/aries-rfcs/tree/bed4989dd6517f7a9de3696800e57e4c6ef49231/features/0434-outofband#invitation-httpsdidcommorgout-of-bandverinvitation) defines `label` as optional.

However 0160: Connection Protocol does not define it as optional.

What do you suggest instead of `''`? I don't think we should put any predefined string there.

Reply:
Make `label` optional in the connection record. It feels weird to store an empty string as the label. This means from now on when using AFJ in a mobile application I would need to check all labels to make sure they don't have the value `''`.

Reply:
I agree that it should be optional in a connection record. But that doesn't solve the problem here. We're trying to build upon existing code and reuse `ConnectionInvitationMessage` where it's not optional. 

Maybe, we should not use `ConnectionInvitationMessage` then.